src/main/java/gov/nasa/jpf/constraints/solvers/ConstraintSolverFactory.java

```diff
@@ -35,6 +35,7 @@
 import java.util.logging.Logger;
 
 /** Factory for instantiating constraint solvers */
+@Deprecated
```

Comment:
I would prefer backing up the ConstraintSolverFactory by the new ConstraintSolverService auto discovery.

All existent code is written against the ConstraintSolverFactory calling the createSolver methods on the root factory. I do not see a meaningful difference in the semantics between the ConstraintSolverService createSolver methods and the ConstraintSolverFactory method. So why should I touch dependent code, only because someone changed the discovery implementation? I am happy to deprecate unnecessary methods in the factory that are no longer required.

Calling `ConstraintSovlerFactory.getRootFactory().createSolver("mySolver")` is exactly the same to calling `ConstraintSolverService.getInstance().createSolver("mySolver")`.

This class *must not* be deprecated unless there is a significant change in the API semantic, IMHO.